### PR TITLE
Secrets Sync UI: Refactor vercel-project destination to expect array from server

### PR DESCRIPTION
**Pull request by hellobontempo**
The [docs](https://developer.hashicorp.com/vault/api-docs/system/secrets-sync#deployment_environments) mentioned `deployment_environments` will be a string but when testing noticed the value returned as an array of strings. Confirmed with the backend team that the type should be updated to an array.

Ember's `changedAttributes()` does not track `array` types, so we transform it to a string to effectively track which keys to include in the `PATCH` payload

### CODE REVIEW

File: ui/app/serializers/sync/destination.js [hunk 1]
```diff
@@ -12,17 +12,14 @@ export default class SyncDestinationSerializer extends ApplicationSerializer {
...
-      // first array element is the old value
-      const [, newValue] = snapshot.changedAttributes()[attr];
-      data[decamelize(attr)] = newValue;
-    }
-    return data;
+    if (snapshot.isNew) return data;
+    // only send changed parameters for PATCH requests
+    const patchData = {};
+    const changedKeys = Object.keys(snapshot.changedAttributes()).map((key) => decamelize(key));
+    changedKeys.forEach((key) => (patchData[key] = data[key]));
```
Comment (hellobontempo) [inline ui/app/serializers/sync/destination.js:21]: this could be pulled into a util for future `PATCH` requests

File: ui/tests/integration/components/sync/secrets/page/destinations/create-and-edit-test.js [hunk 2]
```diff
@@ -212,6 +212,16 @@ module('Integration | Component | sync | Secrets::Page::Destinations::CreateAndE
     gh: ['accessToken'],
     'vercel-project': ['accessToken', 'teamId', 'deploymentEnvironments'],
   };
+  const EXPECTED_VALUE = (key) => {
```
Comment (hellobontempo) [inline ui/tests/integration/components/sync/secrets/page/destinations/create-and-edit-test.js:215]: pulled into method so can be extended in the future for other destination types/attrs

File: ui/app/serializers/sync/destination.js (hunk 1 above)
Comment (hashishaw) [inline ui/app/serializers/sync/destination.js:21]: I think it's probably fine either way, but this could be nice as a reducer function:
```suggestion
    const patchData = changedKeys.reduce((payload, key) => {
    payload[key] = data[key];
    return payload;
    }, {})
```

File: ui/app/transforms/comma-string.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Transform from '@ember-data/serializer/transform';
+
+/**
+ * transforms array types from the server to a comma separated string
+ * useful when using changedAttributes() in the serializer to track attribute changes for PATCH requests
+ * because arrays are not trackable and strings are!
```
Comment (hashishaw) [inline ui/app/transforms/comma-string.js:11]: Great comments 🎉 
Sad ember 😞

**Review by hashishaw — APPROVED** [review]
Excellent! Just one small suggestion 🚀

File: ui/app/serializers/sync/destination.js (hunk 1 above)
Comment (hellobontempo) [reply ui/app/serializers/sync/destination.js:21]: Sure, I can update this. I always think reduce functions are hard to follow so I typically avoid them 🙈

Comment (hashishaw) [reply ui/app/serializers/sync/destination.js:21]: They just take a little getting used to 😁 I started to love them once it clicked
